staging/src/k8s.io/apiserver/pkg/endpoints/installer.go

```diff
@@ -1195,6 +1195,8 @@ func typeToJSON(typeName string) string {
 		return "string"
 	case "v1.IncludeObjectPolicy", "*v1.IncludeObjectPolicy":
 		return "string"
+	case "v1.LogStreamType", "*v1.LogStreamType":
```

Comment:
All of these are "meta/v1" and you are adding "core/v1" - I don't think you need this -- there are LOTS of string-constants not listed here, and this is "generic" apiserver code.

Reply:
https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/127360/pull-kubernetes-integration/1853482480892383232

It seem these changes are necessary, otherwise the apiserver cannot be brought up:

> F1104 17:05:15.235634   23154 openapi.go:43] Failed to build open api spec for root: non-body Restful parameter type should be a simple type, but got : *v1.LogStreamType
!!! [1104 17:05:44] Timed out waiting for apiserver to answer at https://127.0.0.1:6443/healthz; tried 30 waiting 1 between each
!!! [1104 17:05:44] Call tree:
!!! [1104 17:05:44]  1: hack/make-rules/test-cmd.sh:93 kube::util::wait_for_url_with_bearer_token(...)
!!! [1104 17:05:44]  2: hack/make-rules/test-cmd.sh:196 run_kube_apiserver(...)

Reply:
We need to dig into this - I do not think it is reasonable to be enumerating group-specific types in this way.

@jpbetz @deads2k @liggitt @sttts may be able to help?  We have a million fields that are non-primitive types, why is this one special?

Reply:
Digging in, I think it comes via `func (a *APIInstaller) registerResourceHandlers()` calling `podrest.LogREST.NewGetOptions()`.  We have so many interfaces, it's difficult to backtrack.

Reply:
we have very few types that get serialized to query parameters, and I think this came from trying to describe query parameter types in openapi. It *seems* like this information could also be obtained by walking the generated openapi for the type (e.g. schema_k8sio_api_core_v1_PodLogOptions), but I don't know how twisty that would be to obtain / extract here

Reply:
`registerResourceHandlers()` uses the results of `NewGetOptions()` to get `versionedGetOptions` which describes a `v1.PodLogOptions` which includes the `*v1.LogStreamType` field.  I can't tell if that is saved, but it must be that somewhere we try to interpret that.

Reply:
`func (o *openAPI) findCommonParameters()`

Reply:
@liggitt @jpbetz - I suggest to just use `string` and punt this problem - object?

Reply:
No objection, but I'll defer to api-reviewers.